### PR TITLE
Send an empty SETTINGS frame as server preface.

### PR DESCRIPTION
**Pull request by disksing**
As it is described in rfc(https://http2.github.io/http2-spec/#rfc.section.3.5):

> The server connection preface consists of a potentially empty SETTINGS frame (Section 6.5) that MUST be the first frame the server sends in the HTTP/2 connection.

### CODE REVIEW

**Comment by siddontang** [comment]
can we pass all current gRPC test?

**Comment by disksing** [comment]
I have ran './test.sh' and it seems all ok.

**Comment by siddontang** [comment]
Do we need to receive all the client settings and then write the server settings and the following ACK.

**Comment by disksing** [comment]
AFAIU, both server and client can send SettingFrame right after connection establishd.
> SETTINGS parameters are not negotiated; they describe characteristics of the sending peer, which are used by the receiving peer. Different values for the same parameter can be advertised by each peer.

**Comment by siddontang** [comment]
LGTM

But this project lacks tests, we must add more and more later.

File: http2/src/solicit_async.rs [hunk 1]
```diff
@@ -127,17 +127,22 @@ pub fn client_handshake<I : Io + Send + 'static>(conn: I) -> HttpFuture<I> {
...
-    let recv_preface = read_exact(conn, preface_buf)
-        .map_err(|e| e.into())
-        .and_then(|(conn, preface_buf)| {
-            done(if preface_buf == PREFACE {
-                Ok((conn))
-            } else {
-                Err(HttpError::InvalidFrame)
+    let recv_preface = send_preface.and_then(|conn| {
+        read_exact(conn, preface_buf)
+            .map_err(|e| e.into())
```
Comment (overvenus) [inline http2/src/solicit_async.rs:137]: How about `map(Into::into)` ?

**Review by overvenus — APPROVED** [review]
Rest LGTM

File: http2/src/solicit_async.rs (hunk 1 above)
Comment (disksing) [reply http2/src/solicit_async.rs:137]: I suppose they are not very different, and both are equally easy to read.

Comment (overvenus) [reply http2/src/solicit_async.rs:137]: Sure, looks good to me.

**Pull request merged**